platform/src/test/java/org/conscrypt/metrics/MetricsTest.java

```
@@ -0,0 +1,70 @@
+
+public class MetricsTest extends TestCase {
+
+    public static final int TLS_HANDSHAKE_REPORTED = 317;
+
+    // Tests that ReflexiveEvent produces the same event as framework's.
+    public void test_reflexiveEvent() {
+        StatsEvent frameworkStatsEvent = StatsEvent.newBuilder()
+            .setAtomId(TLS_HANDSHAKE_REPORTED)
+            .writeBoolean(false)
+            .writeInt(1) // protocol
+            .writeInt(2) // cipher suite
+            .writeInt(100) // duration
+            .usePooledBuffer()
+            .build();
+
+        ReflexiveStatsEvent reflexiveStatsEvent = ReflexiveStatsEvent.buildEvent(TLS_HANDSHAKE_REPORTED, false, 1, 2, 100);
+        StatsEvent constructedEvent = (StatsEvent) reflexiveStatsEvent.getStatsEvent();
+
+        // TODO(nikitai): Figure out how to use hidden (@hide) getters from StatsEvent
+        // to eliminate the use of reflection
+        try {
+            int fid = (Integer)frameworkStatsEvent.getClass().getMethod("getAtomId").invoke(frameworkStatsEvent);
+            int cid = (Integer)constructedEvent.getClass().getMethod("getAtomId").invoke(constructedEvent);
+            assertEquals(fid, cid);
+
+            int fnb = (Integer)frameworkStatsEvent.getClass().getMethod("getNumBytes").invoke(frameworkStatsEvent);
+            int cnb = (Integer)constructedEvent.getClass().getMethod("getNumBytes").invoke(constructedEvent);
+            assertEquals(fnb, cnb);
+
+            byte[] fbytes = (byte[])frameworkStatsEvent.getClass().getMethod("getBytes").invoke(frameworkStatsEvent);
+            byte[] cbytes = (byte[])constructedEvent.getClass().getMethod("getBytes").invoke(constructedEvent);
+            for (int i = 0; i < fnb; i++) {
+                // skip encoded timestamp (bytes 1-8)
+                if (i < 1 || i > 8) {
+                    assertEquals(fbytes[i], cbytes[i]);
+                }
+            }
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
```

It's better to just let the exception propagate, that way if the test fails there will be a full stack trace rather than just the message. i.e. remove the try/catch and just add `throws Exception` to the test method signature.

---

Done, now test method throws Exception.